registry_module.go

```diff
@@ -572,6 +636,38 @@ func (r *registryModules) Read(ctx context.Context, moduleID RegistryModuleID) (
 
 	return rm, nil
 }
+
+// ReadRegistry fetches a registry module from the Terraform Registry.
+func (r *registryModules) ReadRegistry(ctx context.Context, moduleID RegistryModuleID, version string) (*TerraformRegistryModule, error) {
```

Comment:
Should this be called `ReadByVersion`? It's returning a module but the function is called `ReadRegistry`.

Reply:
The method takes a `version` param, does it read the module or module-version?

A name suggestion would be something like `ReadTerraformRegistryModule`

Reply:
It reads the module.

I will update this to be that name instead!